radio/src/pulses/modules_helpers.h

```diff
@@ -299,8 +299,30 @@ inline bool isModuleSBUS(uint8_t moduleIdx)
   return g_model.moduleData[moduleIdx].type == MODULE_TYPE_SBUS;
 }
 
+inline bool isModuleAFHDS3(uint8_t idx)
+{
+  return g_model.moduleData[idx].type == MODULE_TYPE_AFHDS3;
+}
+
+inline bool isModulePxx2(uint8_t idx){
+  if(idx != EXTERNAL_MODULE) return false;
+#if defined(PXX2)
+  #if defined(HARDWARE_EXTERNAL_MODULE_SIZE_SML)
+     if(g_model.moduleData[idx].type == MODULE_TYPE_XJT_LITE_PXX2 ||
+        g_model.moduleData[idx].type == MODULE_TYPE_R9M_LITE_PRO_PXX2)
+       return true;
+  #endif
+  #if defined(HARDWARE_EXTERNAL_MODULE_SIZE_SML)
+    if(g_model.moduleData[idx].type == MODULE_TYPE_R9M_LITE_PXX2)
+      return true;
+  #endif
+#endif
+  return false;
+}
+
 // order is the same as in enum Protocols in myeeprom.h (none, ppm, pxx, pxx2, dsm, crossfire, multi, r9m, r9m2, sbus)
-static const int8_t maxChannelsModules[] = { 0, 8, 8, 16, -2, 8, 4, 8, 16, 8}; // relative to 8!
+//qba667 count is not matching!
```

Comment:
what does this comment mean?

Reply:
Look at the code in  maxModuleChannels_M8
`return maxChannelsModules[g_model.moduleData[moduleIdx].type];`
It use module type defined as ModuleType
```
MODULE_TYPE_NONE
MODULE_TYPE_PPM,
MODULE_TYPE_XJT_PXX1,
MODULE_TYPE_ISRM_PXX2,
MODULE_TYPE_DSM2,
MODULE_TYPE_CROSSFIRE,
MODULE_TYPE_MULTIMODULE,
MODULE_TYPE_R9M_PXX1,
MODULE_TYPE_R9M_PXX2,
MODULE_TYPE_R9M_LITE_PXX1,
MODULE_TYPE_R9M_LITE_PXX2,
MODULE_TYPE_R9M_LITE_PRO_PXX1,    // Doesn't exist
MODULE_TYPE_R9M_LITE_PRO_PXX2,
MODULE_TYPE_SBUS,
MODULE_TYPE_XJT_LITE_PXX2,
MODULE_TYPE_AFHDS3,
```
The array has to few items to reflect all module types - this should be fixed but as instructed I am not including any necessary changes.